modules/social_features/social_event/modules/social_event_an_enroll/social_event_an_enroll.module

```diff
@@ -0,0 +1,223 @@
+    $params['event_title'] = $node->getTitle();
+    $params['event_url'] = social_event_an_enroll_url($nid, $token);
+
+    $langcode = \Drupal::languageManager()->getCurrentLanguage()->getId();
+    $site_mail = \Drupal::config('system.site')->get('mail');
+
+    $mail = \Drupal::service('plugin.manager.mail')
+      ->mail('social_event_an_enroll', 'social_event_an_enroll', $email, $langcode, $params, $site_mail);
+
+    return empty($mail) ? NULL : $mail['result'];
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function social_event_an_enroll_theme() {
+  return [
+    'event_an_enroll_dialog' => [
+      'variables' => [
+        'links' => NULL,
+      ],
+    ],
+  ];
+}
+
+/**
+ * Override variables for the social page hero data.
+ */
+function social_event_an_enroll_preprocess_page_hero_data(array &$variables) {
+  $account = \Drupal::currentUser();
+
+  // Get current node object or node id.
+  $node = \Drupal::routeMatch()->getParameter('node');
+
+  // Load node object.
+  if (!is_null($node) && !is_object($node)) {
+    $node = Node::load($node);
+  }
+
+  if (is_object($node)) {
+    if (!empty($variables['event_enrollment'])) {
+      if ($node->getType() == 'event') {
+        if ($account->isAnonymous()) {
+
+          $form = \Drupal::formBuilder()
+            ->getForm('Drupal\social_event_an_enroll\Form\EventAnEnrollActionForm');
+
+          $variables['event_enrollment'] = [
+            'enroll_action_form' => $form,
+          ];
+
+          $an_enroll_routes = [
+            'social_event_an_enroll.enroll_dialog',
+            'social_event_an_enroll.enroll_form',
+          ];
+
+          if (in_array(\Drupal::routeMatch()->getRouteName(), $an_enroll_routes)) {
+            unset($variables['event_enrollment']);
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_views_post_render().
+ *
+ * Alter "Event enrollments" views. Add number of anonymous enrollments.
+ */
+function social_event_an_enroll_views_post_render(ViewExecutable $view, &$output, CachePluginBase $cache) {
+  if ($view->id() == 'event_enrollments' && $view->current_display == 'event_enrollments') {
+    if (isset($output['#rows'])) {
+      $nid = \Drupal::routeMatch()->getRawParameter('node');
+      $an_count = social_event_an_enroll_count($nid);
+      if ($an_count && $an_count > 0) {
+        // Fix counter in block title.
+        $view->total_rows += $an_count;
+        // Add default avatar image with counter.
+        if (empty($output['#rows'])) {
+          $output['#rows'][0]['#theme'] = $output['#theme'];
+          $output['#rows'][0]['#view'] = $output['#view'];
+          $output['#rows'][0]['#grouping_level'] = 0;
+          $output['#rows'][0]['#title'] = '';
+        }
+        $output['#rows'][0]['#rows'][] = [
+          '#prefix' => '<div class="avatar">',
+          '#theme' => 'image_style',
+          '#style_name' => 'social_medium',
+          '#uri' => 'public://default-profile-picture.png',
```

Comment:
Hey Taras,

This used to be correct but it will break once the private file system is enabled (such as is the case on the Pachamama platform). We should probably find the default image for the user profile's (because a platform may change this) and use that here. That should fix the issue that PAC is currently having.

Perhaps this can be done by loading the anonymous user and getting the image field value?

~ Alexander

Reply:
@ribel ^

Reply:
@Kingdutch You solution sounds good. I can check it when I will be back to work. Thanks!

Reply:
@ribel Hi Taras, did you get a chance to look at this?

Reply:
@Kingdutch I have fixed it by using existing function `social_profile_get_default_image()`